### DISCUSSION TITLE
M4 not compatible with Ubuntu 24.04

### DISCUSSION THREAD

**Discussion by cfirwin3** [musescore/MuseScore#22616] [General]
I am finding that M4 will not playback properly in Ubuntu (Studio) 24.04 LTS.  The sound stutters in oscillation upon note entry and playback.  The play head moves slowly.
I have found that the Jack Branch PR build does work.

Perhaps there is an incompatibility with the default iterations of Pipewire in the latest Ubuntu release?

Other Alsa and Pulseaudio applications are working fine.

EDIT:
I found that Ubuntu 24.04 Pipewire (at least on Ubuntu Studio) is configured to a 48000 sample rate by default.  Changing this to 44100 allows M4 to work.  I don't think that M4 had a sample rate setting at the moment.  It should.

**Comment by mproffitt** [comment]
My system upgraded to 24.04 this morning and I found I was facing this exact issue.

If you want to keep the higher sample-rate for your system overall and just target MuseScore, you can launch the AppImage with an environment variable for Pipewire that forces the sample-rate down:

```
PIPEWIRE_QUANTUM=256/44100 ./musescore
```

Neither `PIPEWIRE_LATENCY` or `PIPEWIRE_RATE` seemed to work for this but Quantum works perfectly.

https://github.com/PipeWire/pipewire/blob/master/README.md#usage

**Reply by cfirwin3** [reply]
Thanks!

**Reply by alexandre-ficagna** [reply]
It didn't work for me.

**Comment by ghedger** [comment]
MuseScore 4.x (up to 4.65 etc.) isn't compatible with Ubuntu 22.04 either!  The most recent version that works reliably on Linux is MuseScore 2.32.  It seems pretty full-featured.

**Reply by cfirwin3** [reply]
To be fair, 22.04 is nearly 3 LTS versions behind and fast approaching end of life.  For mere engraving, obviously M2-3 remain adequate.  But M4+ offers an enormous amount of features beyond mere engraving and arguably is worth pursuing a more current and longer supported Linux build.  It's one thing to say that software requires switching to Mac/Win, but another to say that current software requires updating one's OS to a current version within brand.  I have had best success with Xubuntu among the Ubuntu variants and I am using 24.04 until the first major update on 26.04, then I will upgrade.

**Comment by MarcSabatella** [comment]
Hmm, I'm not sure what sort of issue you are having with 22.04, but in general it works fine with MU4; tons of users have been using it successfully for years.  If you are having some sort of problem using the current supported version of MuseScore Studio (4.6.5, AppImage build), best to ask for help on the official support forum at musescore.org.  Be sure to describe the problem in detail - what you are doing, what you expect to see happen, what happens instead.

2.3.2 is ancient and definitely not full-featured compared to 4.6.5 - no autoplace, poor spacing, lack of support for tons of features, primitive playback, etc.

**Comment by ghedger** [comment]
I hear what you're saying.  Thing is, when you've got over $1000 invested in VSTs and a realtime DAW environment, you plan upgrades very conservatively.  Pipewire is a must.  MuseScore is only for notation and pitch checking.  Professional mixing and mastering and rendering is done in Bitwig Studio.

**Reply by cfirwin3** [reply]
Sure.  I usually only upgrade after the first major patch on the next LTS... and only when I'm not in the middle of something.
Pipewire is standard for all Ubuntu flavors now (since 24.04 I believe).  So I'm running Pipewire on Xubuntu.
You may have to manually install the Pipewire Jack module (and maybe even Jackd) if your routing and playback system depends on it.

**Comment by ghedger** [comment]
MuseScore 4.6.5 works for me on Ubuntu Linux 24.04.  Just a FYI in line with the discussion from the last week.

**Reply by cfirwin3** [reply]
By the time 26.10 comes out, you should be able to push up to 26.04LTS with a solid release patch in place.  That's about the time I upgrade.  That will give you a long period of support.
